docs/usage/configuration/config_documentation.md

```diff
@@ -3310,6 +3310,12 @@ Options for each entry include:
 
   You might want to disable this if the `subject_claim` returned by the mapping provider is not `sub`.
 
+* `backchannel_logout_is_soft`: by default all OIDC Back-Channel Logouts correspond to hard logouts on
+  the server side. This may not leave users the ability to recover their encryption keys before being logged-out.
+  This can be set to `true` to treat all OIDC Back-Channel logouts as soft-logouts,
+  allowing users to reconnect to the same device if necessary to recover their keys.
+  Defaults to `false`.
```

Comment:
@sandhose Do you have any thoughts on if it makes sense for this to be configurable or not? (Previous conversation at https://github.com/matrix-org/synapse/pull/11414#discussion_r763744469)

The backchannel-logout is used when a user manually logs out of an IdP? Or is it also used when a user's session expires with an IdP?

The former seems like it would make sense with the current (hard-)logout, while the latter should likely be a soft-logout.

Reply:
I think keeping the option to do both is good. Usually, a backchannel logout happens because the user expressed the intent of logging out, but it might also happen when for example an admin kicks out a user to force logging back in, and in this case you might want to do a soft_logout in clients.